vochain/transaction.go

```diff
@@ -174,8 +181,54 @@ func VoteTxCheck(vtx *models.Tx, txBytes, signature []byte, state *State,
 	var vote *models.Vote
 	switch {
 	case process.EnvelopeType.Anonymous:
-		// TODO check snark
-		return nil, fmt.Errorf("snark vote not implemented")
+		// Supports Groth16 proof generated from circom snark compatible
+		// prover
+		proofZkSNARK := ve.Proof.GetZkSnark()
+		if proofZkSNARK == nil {
+			return nil, fmt.Errorf("zkSNARK proof is empty")
+		}
+		proof, publicInputsFromUser, err := zk.ProtobufZKProofToCircomProof(proofZkSNARK)
+		if err != nil {
+			return nil, err
+		}
+
+		verificationKey := app.ZkVks[proofZkSNARK.CircuitParametersIndex]
```

Comment:
check that `proofZkSNARK.CircuitParametersIndex < len(app.ZkVks)` to avoid an easy panic.

Reply:
Right! Forgot about it when finishing the PR, updated